src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -711,6 +711,17 @@ private void GenerateClassInternalsField(Field field)
                 !(@class != null && @class.IsUnion)) || (@class != null && @class.TranslationUnit.IsSystemHeader))
                 return;
 
+            if (field.Class.Fields.Count(c => c.Name.Equals(field.Name)) > 1)
```

Comment:
We need to change atleast one member to keep track of the changes.
So if not OriginalName, should I change 'Name' field?
On 28-May-2015 2:55 AM, "João Matos" <notifications@github.com> wrote:

> In src/Generator/Generators/CSharp/CSharpTextTemplate.cs
> <https://github.com/mono/CppSharp/pull/451#discussion_r31182319>:
>
> > @@ -711,6 +711,17 @@ private void GenerateClassInternalsField(Field field)
> >                  !(@class != null && @class.IsUnion)) || (@class != null && @class.TranslationUnit.IsSystemHeader))
> >                  return;
> >
> > +            if (field.Class.Fields.Count(c => c.Name.Equals(field.Name)) > 1)
>
> We should not modify field.OriginalName. So I'd advise to extract this
> piece of code to a static method and change the
> Helpers.SafeIdentifier(field.OriginalName); call to call the method
> instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mono/CppSharp/pull/451/files#r31182319>.
>